uk.co.compucorp.civicrm.tasksassignments/CRM/Tasksassignments/Upgrader.php

```diff
@@ -793,4 +803,24 @@ private function _uninstallActivityTypes($component, array $types) {
       civicrm_api3('OptionValue', 'delete', array('id' => $id));
     }
   }
+
+  /**
+   * Renames the label for a case status
+   *
+   * @param string $fromLabel
+   * @param string $toLabel
+   */
+  private function _renameCaseStatus($fromLabel, $toLabel) {
+    $optionValue = civicrm_api3('OptionValue', 'getsingle', [
```

Comment:
In general we try to make the upgrader functions idempotent, just in case something goes wrong and we need to run them again. I think because you're doing a `getsingle` call here and using the label in the criteria it will fail the second time. Can you rewrite this so that it could be run multiple times, and does a safety check to ensure the `optionValue` exists before the `create` call? 

It might be at some point in the future the base case statuses are changed in a fresh installation, in that case we wouldn't be sure if this status exists.